spark/src/main/scala/io/substrait/spark/expression/ToSparkExpression.scala

```diff
@@ -68,15 +82,69 @@ class ToSparkExpression(
     Literal(UTF8String.fromString(expr.value()), ToSubstraitType.convert(expr.getType))
   }
 
+  override def visit(expr: SExpression.BinaryLiteral): Literal = {
+    Literal(expr.value().toByteArray, ToSubstraitType.convert(expr.getType))
+  }
+
   override def visit(expr: SExpression.DecimalLiteral): Expression = {
     val value = expr.value.toByteArray
     val decimal = DecimalUtil.getBigDecimalFromBytes(value, expr.scale, 16)
     Literal(Decimal(decimal), ToSubstraitType.convert(expr.getType))
   }
+
   override def visit(expr: SExpression.DateLiteral): Expression = {
     Literal(expr.value(), ToSubstraitType.convert(expr.getType))
   }
 
+  override def visit(expr: SExpression.PrecisionTimestampLiteral): Literal = {
+    Literal(
+      Util.toMicroseconds(expr.value(), expr.precision()),
+      ToSubstraitType.convert(expr.getType))
+
+  }
+
+  override def visit(expr: SExpression.PrecisionTimestampTZLiteral): Literal = {
+    Literal(
+      Util.toMicroseconds(expr.value(), expr.precision()),
+      ToSubstraitType.convert(expr.getType))
+  }
+
+  override def visit(expr: SExpression.IntervalDayLiteral): Literal = {
+    val micros =
+      (expr.days() * Util.SECONDS_PER_DAY + expr.seconds()) * Util.MICROSECOND_PRECISION +
```

Comment:
I think this is wrong? It's equivalent to
```java
(expr.days() * Util.SECONDS_PER_DAY + expr.seconds()) * 6 +
```

Do we need/want something like 
```java
(expr.days() * Util.SECONDS_PER_DAY + expr.seconds()) * Util.MICROSECONDS_PER_SECOND +
```

Reply:
Lol, you're right - and the test didn't fail this as we don't write any "days" or "seconds", just "subseconds". Good catch! Fixed in fd01c2f262534e10e71abf729eac306795df40be